src/transformers/modeling_deberta.py

```diff
@@ -0,0 +1,1233 @@
+        """Constructs a `ModelConfig` from a Python dictionary of parameters."""
+        config = cls()
+        for key, value in json_object.items():
+            if isinstance(value, dict):
+                value = AbsModelConfig.from_dict(value)
+            config.__dict__[key] = value
+        return config
+
+    @classmethod
+    def from_json_file(cls, json_file):
+        """Constructs a `ModelConfig` from a json file of parameters."""
+        with open(json_file, "r", encoding="utf-8") as reader:
+            text = reader.read()
+        return cls.from_dict(json.loads(text))
+
+    def __repr__(self):
+        return str(self.to_json_string())
+
+    def to_dict(self):
+        """Serializes this instance to a Python dictionary."""
+        output = copy.deepcopy(self.__dict__)
+        return output
+
+    def to_json_string(self):
+        """Serializes this instance to a JSON string."""
+
+        def _json_default(obj):
+            if isinstance(obj, AbsModelConfig):
+                return obj.__dict__
+
+        return json.dumps(self.__dict__, indent=2, sort_keys=True, default=_json_default) + "\n"
+
+
+class PoolConfig(AbsModelConfig):
+    """Configuration class to store the configuration of `pool layer`.
+
+    Parameters:
+        config (:class:`~transformers.DeBERTaConfig`): The model config. The field of pool config will be initialized with the `pooling` field in model config.
+
+    Attributes:
+        hidden_size (int): Size of the encoder layers and the pooler layer, default: `768`.
+
+        dropout (float): The dropout rate applied on the output of `[CLS]` token,
+
+        hidden_act (:obj:`str`): The activation function of the projection layer, it can be one of ['gelu', 'tanh'].
+
+    Example::
+        # Here is the content of an example model config file in json format
+
+            {
+              "hidden_size": 768,
+              "num_hidden_layers" 12,
+              "num_attention_heads": 12,
+              "intermediate_size": 3072,
+              ...
+              "pooling": {
+                "hidden_size":  768,
+                "hidden_act": "gelu",
+                "dropout": 0.1
+              }
+            }
+
+    """
+
+    def __init__(self, config=None):
+        """Constructs PoolConfig.
+
+        Args:
+           `config`: the config of the model. The field of pool config will be initialized with the 'pooling' field in model config.
+        """
+
+        self.hidden_size = 768
+        self.dropout = 0
+        self.hidden_act = "gelu"
+        if config:
+            pool_config = getattr(config, "pooling", config)
+            if isinstance(pool_config, dict):
+                pool_config = AbsModelConfig.from_dict(pool_config)
+            self.hidden_size = getattr(pool_config, "hidden_size", config.hidden_size)
+            self.dropout = getattr(pool_config, "dropout", 0)
+            self.hidden_act = getattr(pool_config, "hidden_act", "gelu")
```

Comment:
Can these attributes be in the `DeBERTaConfig` class instead?

Reply:
We use separate config object here is to make the config structured hieratically to make it easier to maintain and read the config file.

Reply:
I would really be in favour of removing that part and adding it to the `DeBERTaConfig` instead, as we do with other models. Keeping a standard API is a very important aspect.